Src/AwesomeAssertions/Primitives/StringEndStrategy.cs

```diff
@@ -19,24 +20,60 @@ public StringEndStrategy(IEqualityComparer<string> comparer, string predicateDes
 
     public void ValidateAgainstMismatch(AssertionChain assertionChain, string subject, string expected)
     {
-        assertionChain
-            .ForCondition(subject!.Length >= expected.Length)
-            .FailWith($"{ExpectationDescription}{{0}}{{reason}}, but {{1}} is too short.", expected, subject);
-
-        if (!assertionChain.Succeeded)
+        if (expected.Length is 0 || comparer.Equals(subject, expected))
         {
             return;
         }
 
-        int indexOfMismatch = subject.Substring(subject.Length - expected.Length).IndexOfFirstMismatch(expected, comparer);
+        var indexes = IndexOfLastMismatch(subject, expected, comparer);
+
+        bool hasMismatch = expected.Length > subject.Length || indexes.index2 >= 0;
 
-        if (indexOfMismatch < 0)
+        if (!hasMismatch)
         {
             return;
         }
 
-        assertionChain.FailWith(
-            $"{ExpectationDescription}{{0}}{{reason}}, but {{1}} differs near {subject.IndexedSegmentAt(indexOfMismatch)}.",
-            expected, subject);
+        var failureMessage = MismatchRenderer.CreateFailureMessage(new MismatchRendererOptions
+        {
+            Subject = subject,
+            Expected = expected,
+            SubjectIndexOfMismatch = Math.Max(indexes.index1, 0),
```

Comment:
When having a mismatch, no one of the indexes must be less than zero, isnt't it?